Feature/664 show melon columns

tomaszguzialek opened the pull request:
There is one issue I am struggling with. When a component is connected via "Link to...", the "Output data streams" appears as the first task pane. Every change in the input columns forces a refresh and a task pane is added at the end. I would like to have it always at the end, but cannot find an easy way to accomplish that.

Screenshot of the overall design
![melons task pane](https://cloud.githubusercontent.com/assets/1483194/10019056/5bcec302-613b-11e5-9ce0-82fc5c07d503.png)
:


Fixes #664

kaspersorensen commented:
Hmm that issue sounds pretty severe so I suggest we find a solution for it. How about always having the task pane, but just hiding/showing it when appropriate? That way we don't add/remove task panes all the time.

Otherwise it looks quite good. Remark on the looks of it: The icon for each stream (next to "Complete rows" and "Incomplete rows" in the screenshot) should be the icon found with IconUtils.OUTPUT_DATA_STREAM_PATH.

tomaszguzialek commented:
Always having a task pane does not solve the problem. I added "Output data streams" task pane in AbstractComponentBuilderPanel as all types of components can produce them, not only analyzers. The always-visible task pane is added in the constructor, so it is always the first task pane being added (so on top). I could try to make every class extending the abstract one reorder the task panes, but that does not sound clean...

tomaszguzialek commented:
Problem solved, @LosD gave me a precious hint. Moved adding of the task pane from the constructor to the end of "init" method, which I have not seen for some strange reason... Icon also fixed, so it is mergeable :).

![melons task pane new icons](https://cloud.githubusercontent.com/assets/1483194/10041866/eb597efc-61e7-11e5-96c2-2b2a9bae691a.png)

LosD commented on desktop/api/src/main/java/org/datacleaner/panels/ColumnListTable.java:
```diff
@@ -77,22 +77,35 @@
     private final WindowContext _windowContext;
     private final boolean _addShadowBorder;
 
+    private boolean _editable;
```
This look like a candidate for final

LosD commented on desktop/api/src/main/java/org/datacleaner/panels/OutputDataStreamsViewer.java:
```diff
@@ -0,0 +1,80 @@
+        for (OutputDataStream outputDataStream : _componentBuilder.getOutputDataStreams()) {
+            setEnabled(true);
+            
+            List<InputColumn<?>> inputColumns = new ArrayList<>();
```
Final

LosD commented on desktop/api/src/main/java/org/datacleaner/panels/OutputDataStreamsViewer.java:
```diff
@@ -0,0 +1,80 @@
+            
+            List<InputColumn<?>> inputColumns = new ArrayList<>();
+            for (Column column : outputDataStream.getTable().getColumns()) {
+                MetaModelInputColumn inputColumn = new MetaModelInputColumn(column);
```
Final... And no reason for the intermediate variable at all, really. No debugging advantage, and it doesn't help the source documenting itself, but if you prefer it like this, keep it (I hate unnecessary intermediate variables in all their forms, but @kaspersorensen loves 'em :))

LosD commented on desktop/api/src/main/java/org/datacleaner/panels/ColumnListTable.java:
```diff
@@ -197,7 +212,7 @@ public void actionPerformed(ActionEvent e) {
         _columnTable.setModel(model);
 
-        if (hasPhysicalColumns) {
+        if ((hasPhysicalColumns) && (_editable)) {
```
The parenthesis seems a bit overkill?

LosD commented:
LGTM except for very minor things.
